monai/data/wsi_reader.py

```diff
@@ -388,34 +424,57 @@ def get_level_count(wsi) -> int:
         """
         return wsi.resolutions["level_count"]  # type: ignore
 
-    @staticmethod
-    def get_size(wsi, level: int) -> Tuple[int, int]:
+    def get_size(self, wsi, level: Optional[int] = None) -> Tuple[int, int]:
         """
         Returns the size (height, width) of the whole slide image at a given level.
 
         Args:
             wsi: a whole slide image object loaded from a file
-            level: the level number where the size is calculated
+            level: the level number where the size is calculated. If not provided the default level (from `self.level`)
+                will be used.
 
         """
+        if level is None:
+            level = self.level
+
         return (wsi.resolutions["level_dimensions"][level][1], wsi.resolutions["level_dimensions"][level][0])
 
-    @staticmethod
-    def get_downsample_ratio(wsi, level: int) -> float:
+    def get_downsample_ratio(self, wsi, level: Optional[int] = None) -> float:
         """
         Returns the down-sampling ratio of the whole slide image at a given level.
 
         Args:
             wsi: a whole slide image object loaded from a file
-            level: the level number where the size is calculated
+            level: the level number where the size is calculated. If not provided the default level (from `self.level`)
+                will be used.
 
         """
+        if level is None:
+            level = self.level
+
         return wsi.resolutions["level_downsamples"][level]  # type: ignore
 
-    def get_file_path(self, wsi) -> str:
+    @staticmethod
+    def get_file_path(wsi) -> str:
         """Return the file path for the WSI object"""
         return str(abspath(wsi.path))
 
+    def get_mpp(self, wsi, level: Optional[int] = None) -> Tuple[float, float]:
+        """
+        Returns the micro-per-pixel resolution of the whole slide image at a given level.
+
+        Args:
+            wsi: a whole slide image object loaded from a file
+            level: the level number where the size is calculated. If not provided the default level (from `self.level`)
+                will be used.
+
+        """
+        if level is None:
+            level = self.level
+
+        factor = float(wsi.resolutions["level_downsamples"][level])
+        return (wsi.metadata["cucim"]["spacing"][1] * factor, wsi.metadata["cucim"]["spacing"][0] * factor)
```

Comment:
I'm not sure whether the spacing here is pointing to level 0 or, but I noticed that he doesn't change with level.

Reply:
Spacing here refers to level 0, and that's why we have to multiply it by downsample.